src/Promitor.Integrations.Sinks.Statsd/StatsdMetricSink.cs

```diff
@@ -31,12 +46,13 @@ public async Task ReportMetricAsync(string metricName, string metricDescription,
             Guard.NotNull(scrapeResult.MetricValues, nameof(scrapeResult.MetricValues));
 
             var reportMetricTasks = new List<Task>();
+            var formatterType = _statsDConfiguration.CurrentValue?.FormatterType ?? StatsdFormatterTypes.DEFAULT;
 
             foreach (var measuredMetric in scrapeResult.MetricValues)
             {
                 var metricValue = measuredMetric.Value ?? 0;
 
-                var reportMetricTask = ReportMetricAsync(metricName, metricDescription, metricValue, new Dictionary<string, string>());
+                var reportMetricTask = _reportMetricsActions[formatterType](metricName, metricDescription, metricValue, scrapeResult.Labels);
```

Comment:
I think we should keep ReportMetric which uses a switch case to call the appropriate method.
```suggestion
                var reportMetricTask = ReportMetric(metricName, metricDescription, metricValue, scrapeResult.Labels);
```

That way we don't need a dictionary to know how to run the code and is simpler to read

Reply:
The current logic from ReportMetricAsync should be moved to a new ReportMetricWithDefaultFormattingAsync

Reply:
I used a dictionary instead of switch statement to follow with open/close principle so we won't have to edit the code if new formatter will be presented, but if you think it won't happen i'll change it to switch statement

Reply:
I'm personally more of a fan of a swich since we'd need to change the code anyway - If you wouldn't mind?